File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const d = {};
    
    for (const name of participant) {
        d[name] = (d[name] || 0) + 1;
    }
    
    for (const name of completion) {
        d[name] -= 1;
    }
    
    for (const name in d) {
        if (d[name] > 0) {
            return name;
        }
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    d = {}
    
    for name in participant:
        d[name] = d.get(name, 0) + 1
    
    for name in completion:
        d[name] -= 1
    
    for name in d:
        if d[name] > 0:
            return name